tensorflow/core/kernels/data/name_utils.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_CORE_KERNELS_DATA_NAME_UTILS_H_
+#define TENSORFLOW_CORE_KERNELS_DATA_NAME_UTILS_H_
+
+#include "tensorflow/core/lib/core/stringpiece.h"
+#include "tensorflow/core/lib/strings/strcat.h"
+#include "tensorflow/core/platform/logging.h"
+#include "tensorflow/core/platform/types.h"
+
+namespace tensorflow {
+namespace data {
+namespace name_utils {
+using namespace tensorflow::strings;
+
+enum DatasetType {
+  RANGE_DATASET = 0,
+  MAP_DATASET = 1,
+};
+
+// Returns the base name of the dataset.
+string ToString(const DatasetType& dataset_type);
+
+// Returns the dataset op name.
+string OpName(const DatasetType& dataset_type);
+
+string DatasetDebugString(const DatasetType& dataset_type,
```

Comment:
do we need this if we have the other one?

Reply:
Tried to combine these two functions into one as below:
```C++
template <typename... Args>
string DatasetDebugString(const DatasetType& dataset_type,
                          const Args&... args) {
  std::initializer_list<StringPiece> arg_list{
      static_cast<const strings::AlphaNum&>(args).Piece()...};

  if (arg_list.size() == 0) {
    return strings::StrCat(OpName(dataset_type), "Op::Dataset");
  }

  string debug_str;
  strings::StrAppend(&debug_str, OpName(dataset_type), "Op(");
  auto iter = arg_list.begin();
  while (iter != arg_list.end() - 1) {
    strings::StrAppend(&debug_str, *iter, ", ");
    ++iter;
  }
  strings::StrAppend(&debug_str, *iter, ")::Dataset");
  return debug_str;
}
```
However, the return output is not expected. For example, for `RangeDataset`, the output changes to `RangeDatasetOp(0, 10, �)::Dataset` rather than `RangeDatasetOp(0, 10, 3)::Dataset`.

@jsimsa Do you have any suggestion?